src/terraform_module_generator/schema/attributes/structure.py

```diff
@@ -0,0 +1,74 @@
+"""
+    :copyright: 2022 Inmanta
+    :contact: code@inmanta.com
+    :license: Inmanta EULA
+"""
+import abc
+import re
+import typing
+
+from inmanta_module_factory.helpers.utils import inmanta_safe_name
+
+from terraform_module_generator.schema import mocks
+from terraform_module_generator.schema.attributes.base import Attribute
+from terraform_module_generator.schema.helpers.cache import cache_method_result
+
+
+class StructureAttribute(Attribute):
+    legacy_regex: re.Pattern[str]
+    regex: re.Pattern[str]
+
+    def __init__(self, path: typing.List[str], schema: typing.Any) -> None:
+        Attribute.__init__(self, path, schema)
+        self.inner_attributes = self.get_inner_attributes(
+            path + [inmanta_safe_name(self.name)], schema
+        )
+
+    @cache_method_result
+    def block_mock(self) -> mocks.BlockMock:
```

Comment:
This method is not used, it I'm not mistaken?

Reply:
Indeed, good catch, thanks!

Reply:
Fixed in https://github.com/inmanta/terraform-generator/pull/40/commits/1fd647de8a916f9bd41cf4c9e77ea3d2f93ce40e